### DISCUSSION TITLE
Restoring JAVA Application Based Window Location Across Multiple Monitors - Problems

### DISCUSSION THREAD

**Discussion by cloves2017** [kangyu-california/PersistentWindows#281] [Q&A]
Hi, I am running windows 10 and have 4 monitors. During the day I run the thinkorswim application (aka tos). I believe the application uses java. The reason I mention that is because PersistentWindows has been the only software that can return the windows layout in the past for this application. But I am still having some issues and maybe its how I use it?

Details: I usually just use 1 monitor when I am not using tos. After I am done working for the day, I goto windows/project and change it from "extend" to "pc screen only". The other 3 monitors shut off. Then in the morning I go back and turn on the "extend". I recently reinstalled PersistentWindows because it had stopped restoring my saved windows for tos on my 4th monitor. That is the one with a ton of small windows lined up for readability. For the first few days that I "did not turn off my other monitors", the app was correctly restoring the tos windows perfectly. Now this is the process I had working. I would open tos first thing in the morning. Then I would hold down "ctrl" and "right restore windows from disk" and load the one I had saved using the ctrl option and named. I wouldn't even open any other application so as not to mess up the windows for tos. BUT somehow either turning the monitors off OR the auto restore option screwed things up again 8( 

I have disabled the "auto restore" but still this morning, the windows wouldn't load correctly into the 4th monitor. 

Anyways I am hoping someone a bit smarter can provide a hint as to how to get things to work consistently.

**Comment by kangyu-california** [comment]
Based on your daily usage model, I would recommend using Restore snapshot feature instead of Restore from disk. The latter is mainly used to recover window layout after PC reboot, which uses fuzzy matching (window process name, class name, window title etc.), and has the limitation of NOT able to capture/restore child window. I guess the 4th monitor has a lot of child windows spawned from TOS app. 

Restore from snapshot uses exact window handle matching and hence the restore is much faster and accurate, also it is capable of restore child windows.

**Reply by cloves2017** [reply]
Thanks kangyu, I'll give that try. It's actually only about 4 windows from tos on that monitor. I'll report back.

**Reply by cloves2017** [reply]
Hi Kangyu, so today I tried the "restore snapshot" on boot up. The "restore snapshot" was not selectable unfortunately. I definitely took snapshots yesterday and gave it a number. I also didn't turn off my other monitors to test. I was able to use "restore windows from disk". I opened up TOS first then used that and it worked. But as mentioned, I didn't turn off the other monitors.

**Reply by cloves2017** [reply]
Just another update. Today the "restore windows from disk" didn't work. Yesterday I closed all my apps and just left TOS open and saved both a snap shot and a named "capture windows to disk". I left all my monitors on all day just to avoid making that a variable. I shutdown and then booted in the am. Opened TOS and then went to PersistentWindows and the "restore snapshot" wasn't selectable. Then holding down ctrl + "restore windows from disk", which gave me a window asking for "enter the name of capture". I had named my capture "1" entered that and nothing. The TOS windows did not pop up into the 4th monitor.

**Reply by kangyu-california** [reply]
snapshots reside in memory, reboot would eliminate them.
Try sleep the PC instead of shutdown.
